File: clients/csharp/src/main/csharp/IO/Swagger/Model/Order.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model
{

    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public class Order : IEquatable<Order>
    {
        
        /// <summary>
        /// Gets or Sets OrderID
        /// </summary>
        [DataMember(Name="orderID", EmitDefaultValue=false)]
        public string OrderID { get; set; }
  
        
        /// <summary>
        /// Gets or Sets ClOrdID
        /// </summary>
        [DataMember(Name="clOrdID", EmitDefaultValue=false)]
        public string ClOrdID { get; set; }
  
        
        /// <summary>
        /// Gets or Sets Account
        /// </summary>
        [DataMember(Name="account", EmitDefaultValue=false)]
        public double? Account { get; set; }
  
        
        /// <summary>
        /// Gets or Sets Symbol
        /// </summary>
        [DataMember(Name="symbol", EmitDefaultValue=false)]
        public string Symbol { get; set; }
  
        
        /// <summary>
        /// Gets or Sets Side
        /// </summary>
        [DataMember(Name="side", EmitDefaultValue=false)]
        public string Side { get; set; }
  
        
        /// <summary>
        /// Gets or Sets SimpleOrderQty
        /// </summary>
        [DataMember(Name="simpleOrderQty", EmitDefaultValue=false)]
        public double? SimpleOrderQty { get; set; }
  
        
        /// <summary>
        /// Gets or Sets OrderQty
        /// </summary>
        [DataMember(Name="orderQty", EmitDefaultValue=false)]
        public double? OrderQty { get; set; }
  
        
        /// <summary>
        /// Gets or Sets Price
        /// </summary>
        [DataMember(Name="price", EmitDefaultValue=false)]
        public double? Price { get; set; }
  
        
        /// <summary>
        /// Gets or Sets MinQty
        /// </summary>
        [DataMember(Name="minQty", EmitDefaultValue=false)]
        public double? MinQty { get; set; }
  
        
        /// <summary>
        /// Gets or Sets StopPx
        /// </summary>
        [DataMember(Name="stopPx", EmitDefaultValue=false)]
        public double? StopPx { get; set; }
  
        
        /// <summary>
        /// Gets or Sets Currency
        /// </summary>
        [DataMember(Name="currency", EmitDefaultValue=false)]
        public string Currency { get; set; }
  
        
        /// <summary>
        /// Gets or Sets SettlCurrency
        /// </summary>
        [DataMember(Name="settlCurrency", EmitDefaultValue=false)]
        public string SettlCurrency { get; set; }
  
        
        /// <summary>
        /// Gets or Sets OrdType
        /// </summary>
        [DataMember(Name="ordType", EmitDefaultValue=false)]
        public string OrdType { get; set; }
  
        
        /// <summary>
        /// Gets or Sets TimeInForce
        /// </summary>
        [DataMember(Name="timeInForce", EmitDefaultValue=false)]
        public string TimeInForce { get; set; }
  
        
        /// <summary>
        /// Gets or Sets ExDestination
        /// </summary>
        [DataMember(Name="exDestination", EmitDefaultValue=false)]
        public string ExDestination { get; set; }
  
        
        /// <summary>
        /// Gets or Sets OrdStatus
        /// </summary>
        [DataMember(Name="ordStatus", EmitDefaultValue=false)]
        public string OrdStatus { get; set; }
  
        
        /// <summary>
        /// Gets or Sets WorkingIndicator
        /// </summary>
        [DataMember(Name="workingIndicator", EmitDefaultValue=false)]
        public bool? WorkingIndicator { get; set; }
  
        
        /// <summary>
        /// Gets or Sets OrdRejReason
        /// </summary>
        [DataMember(Name="ordRejReason", EmitDefaultValue=false)]
        public string OrdRejReason { get; set; }
  
        
        /// <summary>
        /// Gets or Sets SimpleLeavesQty
        /// </summary>
        [DataMember(Name="simpleLeavesQty", EmitDefaultValue=false)]
        public double? SimpleLeavesQty { get; set; }
  
        
        /// <summary>
        /// Gets or Sets LeavesQty
        /// </summary>
        [DataMember(Name="leavesQty", EmitDefaultValue=false)]
        public double? LeavesQty { get; set; }
  
        
        /// <summary>
        /// Gets or Sets SimpleCumQty
        /// </summary>
        [DataMember(Name="simpleCumQty", EmitDefaultValue=false)]
        public double? SimpleCumQty { get; set; }
  
        
        /// <summary>
        /// Gets or Sets CumQty
        /// </summary>
        [DataMember(Name="cumQty", EmitDefaultValue=false)]
        public double? CumQty { get; set; }
  
        
        /// <summary>
        /// Gets or Sets AvgPx
        /// </summary>
        [DataMember(Name="avgPx", EmitDefaultValue=false)]
        public double? AvgPx { get; set; }
  
        
        /// <summary>
        /// Gets or Sets MultiLegReportingType
        /// </summary>
        [DataMember(Name="multiLegReportingType", EmitDefaultValue=false)]
        public string MultiLegReportingType { get; set; }
  
        
        /// <summary>
        /// Gets or Sets Text
        /// </summary>
        [DataMember(Name="text", EmitDefaultValue=false)]
        public string Text { get; set; }
  
        
        /// <summary>
        /// Gets or Sets TransactTime
        /// </summary>
        [DataMember(Name="transactTime", EmitDefaultValue=false)]
        public DateTime? TransactTime { get; set; }
  
        
        /// <summary>
        /// Gets or Sets Timestamp
        /// </summary>
        [DataMember(Name="timestamp", EmitDefaultValue=false)]
        public DateTime? Timestamp { get; set; }
  
        
  
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class Order {\n");
            sb.Append("  OrderID: ").Append(OrderID).Append("\n");
            sb.Append("  ClOrdID: ").Append(ClOrdID).Append("\n");
            sb.Append("  Account: ").Append(Account).Append("\n");
            sb.Append("  Symbol: ").Append(Symbol).Append("\n");
            sb.Append("  Side: ").Append(Side).Append("\n");
            sb.Append("  SimpleOrderQty: ").Append(SimpleOrderQty).Append("\n");
            sb.Append("  OrderQty: ").Append(OrderQty).Append("\n");
            sb.Append("  Price: ").Append(Price).Append("\n");
            sb.Append("  MinQty: ").Append(MinQty).Append("\n");
            sb.Append("  StopPx: ").Append(StopPx).Append("\n");
            sb.Append("  Currency: ").Append(Currency).Append("\n");
            sb.Append("  SettlCurrency: ").Append(SettlCurrency).Append("\n");
            sb.Append("  OrdType: ").Append(OrdType).Append("\n");
            sb.Append("  TimeInForce: ").Append(TimeInForce).Append("\n");
            sb.Append("  ExDestination: ").Append(ExDestination).Append("\n");
            sb.Append("  OrdStatus: ").Append(OrdStatus).Append("\n");
            sb.Append("  WorkingIndicator: ").Append(WorkingIndicator).Append("\n");
            sb.Append("  OrdRejReason: ").Append(OrdRejReason).Append("\n");
            sb.Append("  SimpleLeavesQty: ").Append(SimpleLeavesQty).Append("\n");
            sb.Append("  LeavesQty: ").Append(LeavesQty).Append("\n");
            sb.Append("  SimpleCumQty: ").Append(SimpleCumQty).Append("\n");
            sb.Append("  CumQty: ").Append(CumQty).Append("\n");
            sb.Append("  AvgPx: ").Append(AvgPx).Append("\n");
            sb.Append("  MultiLegReportingType: ").Append(MultiLegReportingType).Append("\n");
            sb.Append("  Text: ").Append(Text).Append("\n");
            sb.Append("  TransactTime: ").Append(TransactTime).Append("\n");
            sb.Append("  Timestamp: ").Append(Timestamp).Append("\n");
            
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as Order);
        }

        /// <summary>
        /// Returns true if Order instances are equal
        /// </summary>
        /// <param name="obj">Instance of Order to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(Order other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.OrderID == other.OrderID ||
                    this.OrderID != null &&
                    this.OrderID.Equals(other.OrderID)
                ) && 
                (
                    this.ClOrdID == other.ClOrdID ||
                    this.ClOrdID != null &&
                    this.ClOrdID.Equals(other.ClOrdID)
                ) && 
                (
                    this.Account == other.Account ||
                    this.Account != null &&
                    this.Account.Equals(other.Account)
                ) && 
                (
                    this.Symbol == other.Symbol ||
                    this.Symbol != null &&
                    this.Symbol.Equals(other.Symbol)
                ) && 
                (
                    this.Side == other.Side ||
                    this.Side != null &&
                    this.Side.Equals(other.Side)
                ) && 
                (
                    this.SimpleOrderQty == other.SimpleOrderQty ||
                    this.SimpleOrderQty != null &&
                    this.SimpleOrderQty.Equals(other.SimpleOrderQty)
                ) && 
                (
                    this.OrderQty == other.OrderQty ||
                    this.OrderQty != null &&
                    this.OrderQty.Equals(other.OrderQty)
                ) && 
                (
                    this.Price == other.Price ||
                    this.Price != null &&
                    this.Price.Equals(other.Price)
                ) && 
                (
                    this.MinQty == other.MinQty ||
                    this.MinQty != null &&
                    this.MinQty.Equals(other.MinQty)
                ) && 
                (
                    this.StopPx == other.StopPx ||
                    this.StopPx != null &&
                    this.StopPx.Equals(other.StopPx)
                ) && 
                (
                    this.Currency == other.Currency ||
                    this.Currency != null &&
                    this.Currency.Equals(other.Currency)
                ) && 
                (
                    this.SettlCurrency == other.SettlCurrency ||
                    this.SettlCurrency != null &&
                    this.SettlCurrency.Equals(other.SettlCurrency)
                ) && 
                (
                    this.OrdType == other.OrdType ||
                    this.OrdType != null &&
                    this.OrdType.Equals(other.OrdType)
                ) && 
                (
                    this.TimeInForce == other.TimeInForce ||
                    this.TimeInForce != null &&
                    this.TimeInForce.Equals(other.TimeInForce)
                ) && 
                (
                    this.ExDestination == other.ExDestination ||
                    this.ExDestination != null &&
                    this.ExDestination.Equals(other.ExDestination)
                ) && 
                (
                    this.OrdStatus == other.OrdStatus ||
                    this.OrdStatus != null &&
                    this.OrdStatus.Equals(other.OrdStatus)
                ) && 
                (
                    this.WorkingIndicator == other.WorkingIndicator ||
                    this.WorkingIndicator != null &&
                    this.WorkingIndicator.Equals(other.WorkingIndicator)
                ) && 
                (
                    this.OrdRejReason == other.OrdRejReason ||
                    this.OrdRejReason != null &&
                    this.OrdRejReason.Equals(other.OrdRejReason)
                ) && 
                (
                    this.SimpleLeavesQty == other.SimpleLeavesQty ||
                    this.SimpleLeavesQty != null &&
                    this.SimpleLeavesQty.Equals(other.SimpleLeavesQty)
                ) && 
                (
                    this.LeavesQty == other.LeavesQty ||
                    this.LeavesQty != null &&
                    this.LeavesQty.Equals(other.LeavesQty)
                ) && 
                (
                    this.SimpleCumQty == other.SimpleCumQty ||
                    this.SimpleCumQty != null &&
                    this.SimpleCumQty.Equals(other.SimpleCumQty)
                ) && 
                (
                    this.CumQty == other.CumQty ||
                    this.CumQty != null &&
                    this.CumQty.Equals(other.CumQty)
                ) && 
                (
                    this.AvgPx == other.AvgPx ||
                    this.AvgPx != null &&
                    this.AvgPx.Equals(other.AvgPx)
                ) && 
                (
                    this.MultiLegReportingType == other.MultiLegReportingType ||
                    this.MultiLegReportingType != null &&
                    this.MultiLegReportingType.Equals(other.MultiLegReportingType)
                ) && 
                (
                    this.Text == other.Text ||
                    this.Text != null &&
                    this.Text.Equals(other.Text)
                ) && 
                (
                    this.TransactTime == other.TransactTime ||
                    this.TransactTime != null &&
                    this.TransactTime.Equals(other.TransactTime)
                ) && 
                (
                    this.Timestamp == other.Timestamp ||
                    this.Timestamp != null &&
                    this.Timestamp.Equals(other.Timestamp)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                
                if (this.OrderID != null)
                    hash = hash * 57 + this.OrderID.GetHashCode();
                
                if (this.ClOrdID != null)
                    hash = hash * 57 + this.ClOrdID.GetHashCode();
                
                if (this.Account != null)
                    hash = hash * 57 + this.Account.GetHashCode();
                
                if (this.Symbol != null)
                    hash = hash * 57 + this.Symbol.GetHashCode();
                
                if (this.Side != null)
                    hash = hash * 57 + this.Side.GetHashCode();
                
                if (this.SimpleOrderQty != null)
                    hash = hash * 57 + this.SimpleOrderQty.GetHashCode();
                
                if (this.OrderQty != null)
                    hash = hash * 57 + this.OrderQty.GetHashCode();
                
                if (this.Price != null)
                    hash = hash * 57 + this.Price.GetHashCode();
                
                if (this.MinQty != null)
                    hash = hash * 57 + this.MinQty.GetHashCode();
                
                if (this.StopPx != null)
                    hash = hash * 57 + this.StopPx.GetHashCode();
                
                if (this.Currency != null)
                    hash = hash * 57 + this.Currency.GetHashCode();
                
                if (this.SettlCurrency != null)
                    hash = hash * 57 + this.SettlCurrency.GetHashCode();
                
                if (this.OrdType != null)
                    hash = hash * 57 + this.OrdType.GetHashCode();
                
                if (this.TimeInForce != null)
                    hash = hash * 57 + this.TimeInForce.GetHashCode();
                
                if (this.ExDestination != null)
                    hash = hash * 57 + this.ExDestination.GetHashCode();
                
                if (this.OrdStatus != null)
                    hash = hash * 57 + this.OrdStatus.GetHashCode();
                
                if (this.WorkingIndicator != null)
                    hash = hash * 57 + this.WorkingIndicator.GetHashCode();
                
                if (this.OrdRejReason != null)
                    hash = hash * 57 + this.OrdRejReason.GetHashCode();
                
                if (this.SimpleLeavesQty != null)
                    hash = hash * 57 + this.SimpleLeavesQty.GetHashCode();
                
                if (this.LeavesQty != null)
                    hash = hash * 57 + this.LeavesQty.GetHashCode();
                
                if (this.SimpleCumQty != null)
                    hash = hash * 57 + this.SimpleCumQty.GetHashCode();
                
                if (this.CumQty != null)
                    hash = hash * 57 + this.CumQty.GetHashCode();
                
                if (this.AvgPx != null)
                    hash = hash * 57 + this.AvgPx.GetHashCode();
                
                if (this.MultiLegReportingType != null)
                    hash = hash * 57 + this.MultiLegReportingType.GetHashCode();
                
                if (this.Text != null)
                    hash = hash * 57 + this.Text.GetHashCode();
                
                if (this.TransactTime != null)
                    hash = hash * 57 + this.TransactTime.GetHashCode();
                
                if (this.Timestamp != null)
                    hash = hash * 57 + this.Timestamp.GetHashCode();
                
                return hash;
            }
        }

    }
}

File: auto-generated/java/src/main/java/io/swagger/client/model/Order.java
package io.swagger.client.model;

import io.swagger.client.StringUtil;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.io.IOException;
import java.math.BigDecimal;
import org.threeten.bp.OffsetDateTime;

import com.google.gson.annotations.SerializedName;


import java.util.Objects;

import io.swagger.annotations.*;



/**
 * Placement, Cancellation, Amending, and History
 **/
@ApiModel(description = "Placement, Cancellation, Amending, and History")
public class Order   {
  
  @SerializedName("orderID")
  private String orderID = null;

  @SerializedName("clOrdID")
  private String clOrdID = null;

  @SerializedName("clOrdLinkID")
  private String clOrdLinkID = null;

  @SerializedName("account")
  private BigDecimal account = null;

  @SerializedName("symbol")
  private String symbol = null;

  @SerializedName("side")
  private String side = null;

  @SerializedName("simpleOrderQty")
  private Double simpleOrderQty = null;

  @SerializedName("orderQty")
  private BigDecimal orderQty = null;

  @SerializedName("price")
  private Double price = null;

  @SerializedName("displayQty")
  private BigDecimal displayQty = null;

  @SerializedName("stopPx")
  private Double stopPx = null;

  @SerializedName("pegOffsetValue")
  private Double pegOffsetValue = null;

  @SerializedName("pegPriceType")
  private String pegPriceType = null;

  @SerializedName("currency")
  private String currency = null;

  @SerializedName("settlCurrency")
  private String settlCurrency = null;

  @SerializedName("ordType")
  private String ordType = null;

  @SerializedName("timeInForce")
  private String timeInForce = null;

  @SerializedName("execInst")
  private String execInst = null;

  @SerializedName("contingencyType")
  private String contingencyType = null;

  @SerializedName("exDestination")
  private String exDestination = null;

  @SerializedName("ordStatus")
  private String ordStatus = null;

  @SerializedName("triggered")
  private String triggered = null;

  @SerializedName("workingIndicator")
  private Boolean workingIndicator = null;

  @SerializedName("ordRejReason")
  private String ordRejReason = null;

  @SerializedName("simpleLeavesQty")
  private Double simpleLeavesQty = null;

  @SerializedName("leavesQty")
  private BigDecimal leavesQty = null;

  @SerializedName("simpleCumQty")
  private Double simpleCumQty = null;

  @SerializedName("cumQty")
  private BigDecimal cumQty = null;

  @SerializedName("avgPx")
  private Double avgPx = null;

  @SerializedName("multiLegReportingType")
  private String multiLegReportingType = null;

  @SerializedName("text")
  private String text = null;

  @SerializedName("transactTime")
  private OffsetDateTime transactTime = null;

  @SerializedName("timestamp")
  private OffsetDateTime timestamp = null;

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getOrderID() {
    return orderID;
  }
  public void setOrderID(String orderID) {
    this.orderID = orderID;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getClOrdID() {
    return clOrdID;
  }
  public void setClOrdID(String clOrdID) {
    this.clOrdID = clOrdID;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getClOrdLinkID() {
    return clOrdLinkID;
  }
  public void setClOrdLinkID(String clOrdLinkID) {
    this.clOrdLinkID = clOrdLinkID;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public BigDecimal getAccount() {
    return account;
  }
  public void setAccount(BigDecimal account) {
    this.account = account;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getSymbol() {
    return symbol;
  }
  public void setSymbol(String symbol) {
    this.symbol = symbol;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getSide() {
    return side;
  }
  public void setSide(String side) {
    this.side = side;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public Double getSimpleOrderQty() {
    return simpleOrderQty;
  }
  public void setSimpleOrderQty(Double simpleOrderQty) {
    this.simpleOrderQty = simpleOrderQty;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public BigDecimal getOrderQty() {
    return orderQty;
  }
  public void setOrderQty(BigDecimal orderQty) {
    this.orderQty = orderQty;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public Double getPrice() {
    return price;
  }
  public void setPrice(Double price) {
    this.price = price;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public BigDecimal getDisplayQty() {
    return displayQty;
  }
  public void setDisplayQty(BigDecimal displayQty) {
    this.displayQty = displayQty;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public Double getStopPx() {
    return stopPx;
  }
  public void setStopPx(Double stopPx) {
    this.stopPx = stopPx;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public Double getPegOffsetValue() {
    return pegOffsetValue;
  }
  public void setPegOffsetValue(Double pegOffsetValue) {
    this.pegOffsetValue = pegOffsetValue;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getPegPriceType() {
    return pegPriceType;
  }
  public void setPegPriceType(String pegPriceType) {
    this.pegPriceType = pegPriceType;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getCurrency() {
    return currency;
  }
  public void setCurrency(String currency) {
    this.currency = currency;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getSettlCurrency() {
    return settlCurrency;
  }
  public void setSettlCurrency(String settlCurrency) {
    this.settlCurrency = settlCurrency;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getOrdType() {
    return ordType;
  }
  public void setOrdType(String ordType) {
    this.ordType = ordType;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getTimeInForce() {
    return timeInForce;
  }
  public void setTimeInForce(String timeInForce) {
    this.timeInForce = timeInForce;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getExecInst() {
    return execInst;
  }
  public void setExecInst(String execInst) {
    this.execInst = execInst;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getContingencyType() {
    return contingencyType;
  }
  public void setContingencyType(String contingencyType) {
    this.contingencyType = contingencyType;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getExDestination() {
    return exDestination;
  }
  public void setExDestination(String exDestination) {
    this.exDestination = exDestination;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getOrdStatus() {
    return ordStatus;
  }
  public void setOrdStatus(String ordStatus) {
    this.ordStatus = ordStatus;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getTriggered() {
    return triggered;
  }
  public void setTriggered(String triggered) {
    this.triggered = triggered;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public Boolean isWorkingIndicator() {
    return workingIndicator;
  }
  public void setWorkingIndicator(Boolean workingIndicator) {
    this.workingIndicator = workingIndicator;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getOrdRejReason() {
    return ordRejReason;
  }
  public void setOrdRejReason(String ordRejReason) {
    this.ordRejReason = ordRejReason;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public Double getSimpleLeavesQty() {
    return simpleLeavesQty;
  }
  public void setSimpleLeavesQty(Double simpleLeavesQty) {
    this.simpleLeavesQty = simpleLeavesQty;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public BigDecimal getLeavesQty() {
    return leavesQty;
  }
  public void setLeavesQty(BigDecimal leavesQty) {
    this.leavesQty = leavesQty;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public Double getSimpleCumQty() {
    return simpleCumQty;
  }
  public void setSimpleCumQty(Double simpleCumQty) {
    this.simpleCumQty = simpleCumQty;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public BigDecimal getCumQty() {
    return cumQty;
  }
  public void setCumQty(BigDecimal cumQty) {
    this.cumQty = cumQty;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public Double getAvgPx() {
    return avgPx;
  }
  public void setAvgPx(Double avgPx) {
    this.avgPx = avgPx;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getMultiLegReportingType() {
    return multiLegReportingType;
  }
  public void setMultiLegReportingType(String multiLegReportingType) {
    this.multiLegReportingType = multiLegReportingType;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getText() {
    return text;
  }
  public void setText(String text) {
    this.text = text;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public OffsetDateTime getTransactTime() {
    return transactTime;
  }
  public void setTransactTime(OffsetDateTime transactTime) {
    this.transactTime = transactTime;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public OffsetDateTime getTimestamp() {
    return timestamp;
  }
  public void setTimestamp(OffsetDateTime timestamp) {
    this.timestamp = timestamp;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Order order = (Order) o;
    return Objects.equals(orderID, order.orderID) &&
        Objects.equals(clOrdID, order.clOrdID) &&
        Objects.equals(clOrdLinkID, order.clOrdLinkID) &&
        Objects.equals(account, order.account) &&
        Objects.equals(symbol, order.symbol) &&
        Objects.equals(side, order.side) &&
        Objects.equals(simpleOrderQty, order.simpleOrderQty) &&
        Objects.equals(orderQty, order.orderQty) &&
        Objects.equals(price, order.price) &&
        Objects.equals(displayQty, order.displayQty) &&
        Objects.equals(stopPx, order.stopPx) &&
        Objects.equals(pegOffsetValue, order.pegOffsetValue) &&
        Objects.equals(pegPriceType, order.pegPriceType) &&
        Objects.equals(currency, order.currency) &&
        Objects.equals(settlCurrency, order.settlCurrency) &&
        Objects.equals(ordType, order.ordType) &&
        Objects.equals(timeInForce, order.timeInForce) &&
        Objects.equals(execInst, order.execInst) &&
        Objects.equals(contingencyType, order.contingencyType) &&
        Objects.equals(exDestination, order.exDestination) &&
        Objects.equals(ordStatus, order.ordStatus) &&
        Objects.equals(triggered, order.triggered) &&
        Objects.equals(workingIndicator, order.workingIndicator) &&
        Objects.equals(ordRejReason, order.ordRejReason) &&
        Objects.equals(simpleLeavesQty, order.simpleLeavesQty) &&
        Objects.equals(leavesQty, order.leavesQty) &&
        Objects.equals(simpleCumQty, order.simpleCumQty) &&
        Objects.equals(cumQty, order.cumQty) &&
        Objects.equals(avgPx, order.avgPx) &&
        Objects.equals(multiLegReportingType, order.multiLegReportingType) &&
        Objects.equals(text, order.text) &&
        Objects.equals(transactTime, order.transactTime) &&
        Objects.equals(timestamp, order.timestamp);
  }

  @Override
  public int hashCode() {
    return Objects.hash(orderID, clOrdID, clOrdLinkID, account, symbol, side, simpleOrderQty, orderQty, price, displayQty, stopPx, pegOffsetValue, pegPriceType, currency, settlCurrency, ordType, timeInForce, execInst, contingencyType, exDestination, ordStatus, triggered, workingIndicator, ordRejReason, simpleLeavesQty, leavesQty, simpleCumQty, cumQty, avgPx, multiLegReportingType, text, transactTime, timestamp);
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class Order {\n");
    
    sb.append("    orderID: ").append(StringUtil.toIndentedString(orderID)).append("\n");
    sb.append("    clOrdID: ").append(StringUtil.toIndentedString(clOrdID)).append("\n");
    sb.append("    clOrdLinkID: ").append(StringUtil.toIndentedString(clOrdLinkID)).append("\n");
    sb.append("    account: ").append(StringUtil.toIndentedString(account)).append("\n");
    sb.append("    symbol: ").append(StringUtil.toIndentedString(symbol)).append("\n");
    sb.append("    side: ").append(StringUtil.toIndentedString(side)).append("\n");
    sb.append("    simpleOrderQty: ").append(StringUtil.toIndentedString(simpleOrderQty)).append("\n");
    sb.append("    orderQty: ").append(StringUtil.toIndentedString(orderQty)).append("\n");
    sb.append("    price: ").append(StringUtil.toIndentedString(price)).append("\n");
    sb.append("    displayQty: ").append(StringUtil.toIndentedString(displayQty)).append("\n");
    sb.append("    stopPx: ").append(StringUtil.toIndentedString(stopPx)).append("\n");
    sb.append("    pegOffsetValue: ").append(StringUtil.toIndentedString(pegOffsetValue)).append("\n");
    sb.append("    pegPriceType: ").append(StringUtil.toIndentedString(pegPriceType)).append("\n");
    sb.append("    currency: ").append(StringUtil.toIndentedString(currency)).append("\n");
    sb.append("    settlCurrency: ").append(StringUtil.toIndentedString(settlCurrency)).append("\n");
    sb.append("    ordType: ").append(StringUtil.toIndentedString(ordType)).append("\n");
    sb.append("    timeInForce: ").append(StringUtil.toIndentedString(timeInForce)).append("\n");
    sb.append("    execInst: ").append(StringUtil.toIndentedString(execInst)).append("\n");
    sb.append("    contingencyType: ").append(StringUtil.toIndentedString(contingencyType)).append("\n");
    sb.append("    exDestination: ").append(StringUtil.toIndentedString(exDestination)).append("\n");
    sb.append("    ordStatus: ").append(StringUtil.toIndentedString(ordStatus)).append("\n");
    sb.append("    triggered: ").append(StringUtil.toIndentedString(triggered)).append("\n");
    sb.append("    workingIndicator: ").append(StringUtil.toIndentedString(workingIndicator)).append("\n");
    sb.append("    ordRejReason: ").append(StringUtil.toIndentedString(ordRejReason)).append("\n");
    sb.append("    simpleLeavesQty: ").append(StringUtil.toIndentedString(simpleLeavesQty)).append("\n");
    sb.append("    leavesQty: ").append(StringUtil.toIndentedString(leavesQty)).append("\n");
    sb.append("    simpleCumQty: ").append(StringUtil.toIndentedString(simpleCumQty)).append("\n");
    sb.append("    cumQty: ").append(StringUtil.toIndentedString(cumQty)).append("\n");
    sb.append("    avgPx: ").append(StringUtil.toIndentedString(avgPx)).append("\n");
    sb.append("    multiLegReportingType: ").append(StringUtil.toIndentedString(multiLegReportingType)).append("\n");
    sb.append("    text: ").append(StringUtil.toIndentedString(text)).append("\n");
    sb.append("    transactTime: ").append(StringUtil.toIndentedString(transactTime)).append("\n");
    sb.append("    timestamp: ").append(StringUtil.toIndentedString(timestamp)).append("\n");
    sb.append("}");
    return sb.toString();
  }
}
